src/stdlib_sorting.fypp

```diff
@@ -0,0 +1,573 @@
+!!   `index` will sort `array` in order of non-increasing values in stable
+!!   order. Otherwise index will sort `array` in order of non-decreasing
+!!   values in stable order.
+!!
+!!#### Examples
+!!
+!! Sorting a related rank one array:
+!!
+!!```Fortran
+!!    subroutine sort_related_data( a, b, work, index, iwork )
+!!        ! Sort `b` in terms or its related array `a`
+!!        integer, intent(inout)           :: a(:)
+!!        integer(int32), intent(inout)    :: b(:) ! The same size as a
+!!        integer(int32), intent(inout)    :: work(:)
+!!        integer(int_size), intent(inout) :: index(:)
+!!        integer(int_size), intent(inout) :: iwork(:)
+!!    ! Find the indices to sort a
+!!        call sort_index(a, index(1:size(a)),&
+!!            work(1:size(a)/2), iwork(1:size(a)/2))
+!!    ! Sort b based on the sorting of a
+!!        b(:) = b( index(1:size(a)) )
+!!    end subroutine sort_related_data
+!!```
+!!
+!! Sorting a rank 2 array based on the data in a column
+!!
+!!```Fortran
+!!    subroutine sort_related_data( array, column, work, index, iwork )
+!!    ! Sort `a_data` in terms or its component `a`
+!!        integer, intent(inout)           :: a(:,:)
+!!        integer(int32), intent(in)       :: column
+!!        integer(int32), intent(inout)    :: work(:)
+!!        integer(int_size), intent(inout) :: index(:)
+!!        integer(int_size), intent(inout) :: iwork(:)
+!!        integer, allocatable             :: dummy(:)
+!!        integer :: i
+!!        allocate(dummy(size(a, dim=1)))
+!!    ! Extract a component of `a_data`
+!!        dummy(:) = a(:, column)
+!!    ! Find the indices to sort the column
+!!        call sort_index(dummy, index(1:size(dummy)),&
+!!                        work(1:size(dummy)/2), iwork(1:size(dummy)/2))
+!!    ! Sort a based on the sorting of its column
+!!        do i=1, size(a, dim=2)
+!!            a(:, i) = a(index(1:size(a, dim=1)), i)
+!!        end do
+!!    end subroutine sort_related_data
+!!```
+!!
+!! Sorting an array of a derived type based on the dsta in one component
+!!```fortran
+!!    subroutine sort_a_data( a_data, a, work, index, iwork )
+!!    ! Sort `a_data` in terms or its component `a`
+!!        type(a_type), intent(inout)      :: a_data(:)
+!!        integer(int32), intent(inout)    :: a(:)
+!!        integer(int32), intent(inout)    :: work(:)
+!!        integer(int_size), intent(inout) :: index(:)
+!!        integer(int_size), intent(inout) :: iwork(:)
+!!    ! Extract a component of `a_data`
+!!        a(1:size(a_data)) = a_data(:) % a
+!!    ! Find the indices to sort the component
+!!        call sort_index(a(1:size(a_data)), index(1:size(a_data)),&
+!!                        work(1:size(a_data)/2), iwork(1:size(a_data)/2))
+!!    ! Sort a_data based on the sorting of that component
+!!        a_data(:) = a_data( index(1:size(a_data)) )
+!!    end subroutine sort_a_data
+!!```
+
+#:for k1 in INT_KINDS
+    public ${k1}$_ord_sort
```

Comment:
Why are all these specific procedures `public`? Is the `public` `interface` `ord_sort` not enough?

Reply:
My original `test_sorting.f90` passed  the sorting procedure as an argument to a testing procedure. That require that the specific procedures, at least for `int32`, be available. As the API changed the sorting procedure argument to the testing procedure became impractical, and I no longer needed the specific procedures. I have removed them from the public interface in both `stdlib_sorting.fypp` and `stdlib_sorting.md`.